### PR TITLE
GEOPY-847

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-847 - Clip by extent handles 2D input**

### CODE REVIEW

File: geoh5py/shared/utils.py [hunk 1]
```diff
@@ -374,7 +374,7 @@ def mask_by_extent(
     """
     Find indices of locations within a rectangular extent.
 
-    :param locations: shape(*, 3) Coordinates to be evaluated.
+    :param locations: shape(*, 3) or shape(*, 3) Coordinates to be evaluated.
```
Comment (benk-mira) [inline geoh5py/shared/utils.py:377]:
```suggestion
    :param locations: shape(*, 3) or shape(*, 2) Coordinates to be evaluated.
```

File: geoh5py/shared/utils.py [hunk 2]
```diff
@@ -385,25 +385,17 @@ def mask_by_extent(
...
-    indices = np.all(
-        np.c_[
-            np.all(locations >= extent[0, :], axis=1),
-            np.all(locations <= extent[1, :], axis=1),
-        ],
-        axis=1,
-    )
+    indices = np.ones(locations.shape[0], dtype=bool)
+    for loc, lim in zip(locations.T, extent.T):
+        indices = np.logical_and(indices, np.logical_and(lim[0] <= loc, loc <= lim[1]))
```
Comment (benk-mira) [inline geoh5py/shared/utils.py:398]: Is there a reason to prefer the numpy method here?  How about:

`indices &= (lim[0] <= loc) & (loc <= lim[1])`

**Review by benk-mira — COMMENTED** [review]
Couple of suggestions

File: geoh5py/shared/utils.py (hunk 2 above)
Comment (domfournier) [reply geoh5py/shared/utils.py:398]: No good reason, changing.

Comment (benk-mira) [reply geoh5py/shared/utils.py:398]: did you forget to push?

**Review by benk-mira — APPROVED** [review]
Rad

**Pull request merged**